RecFCCeeCalorimeter/src/components/CellPositionsHCalPhiThetaSegTool.cpp

```diff
@@ -0,0 +1,211 @@
+#include "CellPositionsHCalPhiThetaSegTool.h"
+
+#include "edm4hep/CalorimeterHitCollection.h"
+#include <DDRec/DetectorData.h>
+
+using dd4hep::DetElement;
+
+DECLARE_COMPONENT(CellPositionsHCalPhiThetaSegTool)
+
+CellPositionsHCalPhiThetaSegTool::CellPositionsHCalPhiThetaSegTool(
+    const std::string &type,
+    const std::string &name,
+    const IInterface *parent)
+    : GaudiTool(type, name, parent)
+{
+  declareInterface<ICellPositionsTool>(this);
+}
+
+StatusCode CellPositionsHCalPhiThetaSegTool::initialize()
+{
+  StatusCode sc = GaudiTool::initialize();
+  if (sc.isFailure())
+    return sc;
+  m_geoSvc = service("GeoSvc");
+  if (!m_geoSvc)
+  {
+    error() << "Unable to locate Geometry service." << endmsg;
+    return StatusCode::FAILURE;
+  }
+  // get PhiTheta segmentation
+  m_segmentation = dynamic_cast<dd4hep::DDSegmentation::FCCSWGridPhiTheta_k4geo *>(
+      m_geoSvc->getDetector()->readout(m_readoutName).segmentation().segmentation());
+  if (m_segmentation == nullptr)
+  {
+    error() << "There is no phi-theta segmentation!!!!" << endmsg;
+    // return StatusCode::FAILURE;
```

Comment:
why there's an error message here not followed by a return StatusCode::FAILURE? Did you forget uncommenting the line or there is a good reason? In the latter case, you should not use an error(), rather a warning(), and handle the else accordingly

Reply:
this was not intentional, just a copy-paste from the CellPositionsHCalBarrelPhiThetaSegTool. In the new version of the code, it will return status code failure.